src/main/shell-sync.ts

```diff
@@ -1,18 +1,34 @@
 import shellEnv = require("shell-env")
 import logger from "./logger"
+import * as os from "os";
 
-export async function shellSync() {
-  const env = await shellEnv()
+interface Env {
+  [key: string]: string;
+}
+
+/**
+ * shellSync loads what would have been the environment if this application was
+ * run from the command line, into the process.env object. This is especially
+ * useful on macos where this always needs to be done.
+ * @param locale Should be electron's `app.getLocale()`
+ */
+export function shellSync(locale: string) {
+  const { shell } = os.userInfo();
+  const env: Env = JSON.parse(JSON.stringify(shellEnv.sync(shell)))
+  if (!env.LANG) {
+    // the LANG env var expects an underscore instead of electron's dash
+    env.LANG = `${locale.replace('-', '_')}.UTF-8`;
+  } else if (!env.LANG.endsWith(".UTF-8")) {
+    env.LANG += ".UTF-8"
+  }
 
   // Overwrite PATH on darwin
   if (process.env.NODE_ENV === "production" && process.platform === "darwin") {
     process.env["PATH"] = env.PATH
   }
 
-  let key = null
-  for(key in env) {
-    if(!env.hasOwnProperty(key) || process.env[key]) continue // skip existing and prototype keys
```

Comment:
@kke do you remember why did we have this check?

Reply:
Yes. The loop will go through all of the JS prototype object's functions and what ever without that. This check will limit the loop to only go  through the actual user defined attributes.

And the `|| process.env[key]` is to avoid overwriting.

Reply:
I think we want to explicitly overwrite `LANG` here. Maybe it could be done similar way as `PATH`?

Reply:
Maybe. But it could also have been done stupidly orginally :)

Reply:
Given this fiddle: https://jsfiddle.net/2v3pdbqL/

I think that I should have put `...env` first because JS overwrites the first's fields when joining.